### PR TITLE
feat: ugprade aws-sdk-go to support IRSA in k8s

### PR DESCRIPTION
**Pull request by adnankobir**
Upgrades the aws sdk to support IRSA - https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html

### CODE REVIEW

**Comment by adnankobir** [comment]
@bradrydzewski @tboerger review would be much appreciated

File: plugin.go [hunk 1]
```diff
@@ -104,7 +104,15 @@ func (p *Plugin) Exec() error {
 		log.Warn("AWS Key and/or Secret not provided (falling back to ec2 instance profile)")
 	}
 
-	client := s3.New(session.New(), conf)
+	sess, err := session.NewSession(conf)
+	if err != nil {
+		log.WithFields(log.Fields{
```
Comment (donny-dont) [inline plugin.go:109]: If its a single field you can do `log.WithField`

**Review by donny-dont — APPROVED** [review]

File: plugin.go (hunk 1 above)
Comment (tboerger) [reply plugin.go:109]: It should not require `WithField` or `WithFields`, just use `WithError(err)`.

**Comment by brennoo** [comment]
can we merge this PR ?

**Comment by FutureSharks** [comment]
Yes please!

**Comment by shubham149** [comment]
Added the support as part of https://github.com/drone-plugins/drone-s3/pull/80

**Pull request closed without merging**